### ISSUE TITLE
ENH: Optimize fusion strategy

### ISSUE THREAD

**Issue by aresnow1** [xorbitsai/xorbits#384]
Currently graph fusion picks two initial nodes to merge into one subtask, which is not efficient for some embarrassingly parallel tasks.

![9f7d5d77-b990-442a-a1b0-ae39c183d213](https://user-images.githubusercontent.com/109642806/232770911-c3a8932b-22af-4c26-a761-8519e0c1c8f4.jpeg)

As shown above, there is no need to merge those two nodes, as they can run in parallel in different slots.
